SocketTimeoutException

I keep getting `SocketTimeoutException` exceptions. 
Any ideas what potentially causes this? 
On a similar note, how can I set the timeout parameters? (if they are exposed to the java api). 

```
Exception in thread "main" java.net.SocketTimeoutException: timeout
	at okio.Okio$3.newTimeoutException(Okio.java:207)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:261)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:215)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:306)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:300)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:196)
	at com.squareup.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:186)
	at com.squareup.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:127)
	at com.squareup.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:737)
	at com.squareup.okhttp.internal.http.HttpEngine.access$200(HttpEngine.java:87)
	at com.squareup.okhttp.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:722)
	at com.squareup.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:576)
	at com.squareup.okhttp.Call.getResponse(Call.java:287)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.execute(Call.java:80)
	at io.minio.MinioClient.execute(MinioClient.java:699)
	at io.minio.MinioClient.executePut(MinioClient.java:949)
	at io.minio.MinioClient.executePut(MinioClient.java:973)
	at io.minio.MinioClient.putObject(MinioClient.java:1934)
	at io.minio.MinioClient.putObject(MinioClient.java:1956)
	at io.minio.MinioClient.putObject(MinioClient.java:1840)
	at Datastore.publishFile(Datastore.java:113)
	at Datastore.publishFile(Datastore.java:91)
	at Main.main(Main.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:170)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.Okio$2.read(Okio.java:139)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:211)
	... 27 more
```

--- a/src/main/java/io/minio/MinioClient.java
+++ b/src/main/java/io/minio/MinioClient.java
@@ -47,6 +47,7 @@ import java.net.URL;
 import java.security.NoSuchAlgorithmException;
 import java.security.InvalidKeyException;
 import java.util.*;
+import java.util.concurrent.TimeUnit;
 import java.nio.charset.StandardCharsets;
 import java.nio.channels.Channels;
 import java.nio.file.Path;
@@ -55,7 +56,6 @@ import java.nio.file.Files;
 import java.nio.file.StandardOpenOption;
 import com.google.common.io.ByteStreams;
 import java.nio.file.StandardCopyOption;
-
 
 /**
  * <p>
@@ -511,6 +511,23 @@ public final class MinioClient {
     }
   }
 
+
+  /**
+   * Sets HTTP connect, write and read timeouts.  A value of 0 means no timeout, otherwise values must be between 1 and
+   * Integer.MAX_VALUE when converted to milliseconds.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code minioClient.setTimeout(TimeUnit.SECONDS.toMillis(10), TimeUnit.SECONDS.toMillis(10), TimeUnit.SECONDS.toMillis(30)); }</pre>
+   *
+   * @param connectTimeout    HTTP connect timeout in milliseconds.
+   * @param writeTimeout      HTTP write timeout in milliseconds.
+   * @param readTimeout       HTTP read timeout in milliseconds.
+   */
+  public void setTimeout(long connectTimeout, long writeTimeout, long readTimeout) {
+    httpClient.setConnectTimeout(connectTimeout, TimeUnit.MILLISECONDS);
+    httpClient.setWriteTimeout(writeTimeout, TimeUnit.MILLISECONDS);
+    httpClient.setReadTimeout(readTimeout, TimeUnit.MILLISECONDS);
+  }
 
   /**
    * Creates Request object for given request parameters.

